6_nodejs/week3/app.js

```diff
@@ -0,0 +1,57 @@
+import knex from 'knex'
+const knexInstance = knex({
+  client: "mysql2",
+  connection: {
+    host: process.env.DB_HOST || "127.0.0.1",
+    port: process.env.DB_PORT || 3306,
+    user: process.env.DB_USER || "root",
+    password: process.env.DB_PASSWORD || "my-secret-pw",
+    database: process.env.DB_NAME || "hyf_node_week3_warmup",
+    multipleStatements: true,
+  },
+});
+
+import 'dotenv/config'
+import express from 'express'
+const app = express();
+const port = process.env.PORT || 3000;
+
+app.use(express.json());
+
+const apiRouter = express.Router();
+app.use("/api", apiRouter);
+
+const contactsAPIRouter = express.Router();
+apiRouter.use("/contacts", contactsAPIRouter);
+
+contactsAPIRouter.get("/", async (req, res) => {
+  let query = knexInstance.select("*").from("contacts");
+
+  if ("sort" in req.query) {
+    const orderBy = req.query.sort;
+    const allowedColumnsForSort = [
+        "first_name",
+        "last_name",
+        "email",
+        "phone",
+        "id"
+      ];
```

Comment:
You decided to fix the SQL injection vulnerability via [allowlisting the parameters](https://planetscale.com/blog/how-to-prevent-sql-injection-attacks-in-node-js#allowlisting). It is a great way to handle user input, since as we know, user input should **NEVER** be trusted. 🧠